### PR TITLE
Remove the TypeChecker interface from Evaluator

### PR DESCRIPTION
**Pull request by ozevren**
+ Also, the PredicateEvaluator interface is subsumed to the Evaluator interface.

**What this PR does / why we need it**:
This is a sanity-refactoring that teases out responsibilities for the interfaces. With this CL, the exact nature of dependencies to the expression language is clearly demarcated.

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #

**Special notes for your reviewer**:

**Release note**:

```release-note
NONE
```

### CODE REVIEW

File: mixer/pkg/runtime/controller.go [hunk 1]
```diff
@@ -114,7 +115,7 @@ func (c *Controller) publishSnapShot() {
 	// attribute manifests are used by type inference during handler creation.
 	attributes := c.processAttributeManifests()
 
-	if cl, ok := c.eval.(VocabularyChangeListener); ok {
+	if cl, ok := c.evaluator.(VocabularyChangeListener); ok {
```
Comment (ozevren) [inline mixer/pkg/runtime/controller.go]: cc: @mandarjog 

Mandar, can you please take a look at this and tell me if this is sane?

**Comment by ozevren** [comment]
Friendly ping. Squash and merge looks pretty tempting.

**Comment by geeknoid** [comment]
Hmm, I thought I approved this already...

/lgtm

File: mixer/pkg/runtime/controller.go (hunk 1 above)
Comment (mandarjog) [reply mixer/pkg/runtime/controller.go]: Lgtm

**Pull request merged**